integrations/opennms-vmware/src/main/java/org/opennms/protocols/vmware/VmwareViJavaAccess.java

```diff
@@ -202,7 +210,26 @@ public VmwareViJavaAccess(String hostname) throws MarshalException, ValidationEx
     public void connect() throws MalformedURLException, RemoteException {
         relax();
 
-        m_serviceInstance = new ServiceInstance(new URL("https://" + m_hostname + "/sdk"), m_username, m_password);
+	//Fix for NMS-8204: use singleton of serverConnection, also make method synchronized
+	String connectionString = new String(m_hostname + m_username + m_password);
+	if(m_userSessions.get(connectionString) == null)
```

Comment:
Are the sessions thread safe? What happens if two different threads try and the same session?

Reply:
Shouldn't this block/method be "synchronized"?